### ISSUE TITLE
add prop `status: 'running' | 'ready' | 'rejected'` to root render prop

### ISSUE THREAD

**Issue by kitos** [kuzn-ilya/react-promise-renderer#7]
I believe it would be handy in some cases use one flag instead of several

**Comment by kuzn-ilya** [comment]
I agree, it makes sence. But there is no direct mapping between boolean flags like running and proposed statuses. E.g. it's possible to have ready-to-use result and to execute promise-returning function at the same time. What should be returned as a status in such a case?

**Comment by kuzn-ilya** [comment]
@kitos, Are you still believe that proposed change will be useful?

**Comment by kitos** [comment]
Sorry for not responding for a long time 😄 
Yes I still believe it would be nice to have a render prop with api similar (or even qual) to [react-apollo Query](https://www.apollographql.com/docs/react/api/react-apollo.html#query).
